File: web/src/types/team.ts
import { User } from "@/types/user";

export interface Team {
    id?: number;
    name?: string;
    description?: string;
    email?: string;
    captain_id?: number;
    captain?: User;
    is_locked?: boolean;
    created_at?: string;
    updated_at?: string;
    users?: Array<User>;
}

export interface TeamFindRequest {
    id?: number;
    name?: string;
    captain_id?: number;
    page?: number;
    size?: number;
    sort_key?: string;
    sort_order?: string;
}

export interface TeamUpdateRequest {
    id?: number;
    name?: string;
    description?: string;
    email?: string;
    captain_id?: number;
    is_locked?: boolean;
}

export interface TeamCreateRequest {
    name: string;
    description: string;
    email?: string;
    captain_id: number;
}

export interface TeamDeleteRequest {
    id: number;
}

export interface TeamJoinRequest {
    id: number;
    token: string;
}

export interface TeamLeaveRequest {
    id: number;
}

export interface TeamGetInviteTokenRequest {
    id: number;
}

export interface TeamUpdateInviteTokenRequest {
    id: number;
}

export interface TeamDeleteUserRequest {
    id: number;
    user_id: number;
}

File: internal/model/request/team.go
package request

type TeamCreateRequest struct {
	Name        string `binding:"required" json:"name"`
	Description string `json:"description"`
	Email       string `json:"email"`
	CaptainId   uint   `binding:"required" json:"captain_id"`
}

type TeamUpdateRequest struct {
	ID          uint   `json:"-"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Email       string `json:"email"`
	CaptainId   uint   `json:"captain_id"`
	IsLocked    *bool  `json:"is_locked"`
}

type TeamFindRequest struct {
	ID        uint   `json:"id" form:"id"`
	Name      string `json:"name" form:"name"`
	UserID    *uint  `json:"user_id" form:"user_id"`
	CaptainID uint   `json:"captain_id" form:"captain_id"`
	GameID    *uint  `json:"game_id" form:"game_id"`
	Page      int    `json:"page" form:"page"`
	Size      int    `json:"size" form:"size"`
	SortKey   string `json:"sort_key" form:"sort_key"`
	SortOrder string `json:"sort_order" form:"sort_order"`
}

type TeamDeleteRequest struct {
	ID uint `json:"-"`
}

type TeamGetInviteTokenRequest struct {
	ID     uint `json:"-"`
	UserID uint `json:"-"`
}

type TeamUpdateInviteTokenRequest struct {
	ID     uint `json:"-"`
	UserID uint `json:"-"`
}
